frontend/src/Pages/ResourcesManagement.tsx

```diff
@@ -20,55 +19,76 @@ import DeleteForm from '../Components/forms/DeleteForm';
 import { useDebounceValue } from 'usehooks-ts';
 import ExternalLink from '@/Components/ExternalLink';
 import ULIComponent from '@/Components/ULIComponent.tsx';
-import { useToast } from '@/Context/ToastCtx';
-import API from '@/api/api';
+
+interface ToastProps {
+    state: ToastState;
+    message: string;
+}
 
 type EditableResourceCollection = ResourceCategory & {
     isModified: boolean;
 };
-
 export default function ResourcesManagement() {
-    const { toaster } = useToast();
-    const { data, error, mutate, isLoading } = useSWR<
-        ServerResponseMany<ResourceCategory>,
-        AxiosError
-    >('/api/left-menu');
+    //eslint-disable-next-line
+    const { data, error, mutate, isLoading } = useSWR('/api/left-menu');
 
     const [collectionList, setCollectionList] = useState<
         EditableResourceCollection[]
     >([]);
+    const [initialCollectionList, setInitialCollectionList] = useState<
+        EditableResourceCollection[]
+    >([]);
     const [collectionToDelete, setCollectionToDelete] = useState<
         number | undefined
     >();
     const [selectedCollectionIndex, setSelectedCollectionIndex] = useState<
         number | undefined
-    >(undefined);
+    >();
+    const [originalLinkValues, setOriginalLinkValues] = useState<
+        Record<number, ResourceLink>
+    >({});
     const [hasDeletedCollection, setHasDeletedCollection] = useState(false);
+    const [, setToast] = useState<ToastProps>({
+        state: ToastState.null,
+        message: ''
+    });
 
     const addCollectionModal = useRef<HTMLDialogElement>(null);
     const deleteCollectionModal = useRef<HTMLDialogElement>(null);
 
     useEffect(() => {
         if (data) {
+            //eslint-disable-next-line
             const updatedData = data.data.map(
-                (collection: ResourceCategory, idx: number) => {
-                    return {
-                        ...collection,
-                        id: idx,
-                        isModified: false
-                    } as EditableResourceCollection;
-                }
+                (collection: EditableResourceCollection) => ({
+                    ...collection,
+                    id: Math.random(),
+                    isModified: false
+                })
             );
+
             setHasDeletedCollection(false);
             setSelectedCollectionIndex(undefined);
+            //eslint-disable-next-line
             setCollectionList(updatedData);
+            //eslint-disable-next-line
```

Comment:
Can we resolve this so `estlint-disable-next-line` isn't necessary? If we need it, please include a comment explaining why we're disabling eslint.